editor/src/components/editor/store/dispatch-strategies.tsx

```diff
@@ -109,6 +109,14 @@ export function interactionFinished(
       newStrategyState: withClearedSession,
     }
   } else {
+    // Try to keep any updated metadata that may have been populated into here
+    // in the meantime.
+    newEditorState = {
```

Comment:
sorry for the nitpick, but would you mind declaring a new const here? the `let newEditorState` in an almost 100 line long function takes a big toll on me when reading / refactoring